src/API/Google/MerchantMetrics.php

```diff
@@ -0,0 +1,83 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query\MerchantFreeListingReportQuery;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WP;
+use DateTime;
+use Google\Service\ShoppingContent;
+use Google\Service\ShoppingContent\SearchResponse;
+
+/**
+ * Class MerchantMetrics
+ *
+ * @since   x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Google
+ */
+class MerchantMetrics implements OptionsAwareInterface {
+
+	use OptionsAwareTrait;
+
+	/**
+	 * The shopping service.
+	 *
+	 * @var ShoppingContent
+	 */
+	protected $service;
+
+	/**
+	 * @var WP
+	 */
+	protected $wp;
+
+	protected const MAX_QUERY_START_DATE = '2020-01-01';
+
+	/**
+	 * MerchantMetrics constructor.
+	 *
+	 * @param ShoppingContent $service
+	 * @param WP              $wp
+	 */
+	public function __construct( ShoppingContent $service, WP $wp ) {
+		$this->service = $service;
+		$this->wp      = $wp;
+	}
+
+	/**
+	 * Get number of free listing clicks.
+	 *
+	 * @return int
+	 */
+	public function get_free_listing_clicks(): int {
+		// Google API requires a date clause to be set but there doesn't seem to be any limits on how wide the range
+		$query = ( new MerchantFreeListingReportQuery( [] ) )
+			->set_client( $this->service, $this->options->get_merchant_id() )
+			->where_date_between( self::MAX_QUERY_START_DATE, $this->get_today() )
+			->fields( [ 'clicks' ] );
+
+		/** @var SearchResponse $response */
+		$response = $query->get_results();
+
+		if ( empty( $response ) || empty( $response->getResults() ) ) {
+			return 0;
+		}
+
+		$report_row = $response->getResults()[0];
+
+		return (int) $report_row->getMetrics()->getClicks();
+	}
+
+	/**
+	 * Get today's date.
+	 *
+	 * @return string
+	 */
+	protected function get_today(): string {
```

Comment:
I wonder if this would be useful in other places, for example:
https://github.com/woocommerce/google-listings-and-ads/blob/bd62aaed77f591e3d7f69d6b9b01625adb5c0eca/src/API/Google/Proxy.php#L649-L655
(Maybe with an indication of the format of the return string)

Reply:
In that example a different timezone is used. I think there is definitely room for some datetime helper code in this project. I played around with some ideas while working on this PR but didn't arrive on anything.

Reply:
> In that example a different timezone is used. 

❓  Sorry, end of the week, couldn't quite parse that. 

But I'd agree that some datetime helper code would be good, maybe add it as an issue with some notes about what you were thinking and a reference to this method for refactoring?